src/features/courses/course-route.ts

```diff
@@ -1117,4 +1118,67 @@ router.delete('/attachments/:id',
         next(httpResponse.Ok('Attachment deleted successfully', result));
     }));
 
+router.post('/question/editor/save',
+    authenticationMiddleware,
+    validate(saveQuestionValidation),
+    // This is due to a typescript issue where the type mismatches extractMap
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    asyncHandler(async (req: RederlyExpressRequest<SaveQuestionRequest.params, unknown, SaveQuestionRequest.body, SaveQuestionRequest.query>, _res: Response, next: NextFunction) => {
+        if (_.isNil(req.session)) {
+            throw new Error(Constants.ErrorMessage.NIL_SESSION_MESSAGE);
+        }
+
+        const writeFilePath = urljoin(`private/my/${req.session.userId}`, req.body.relativePath);
+        const result = await rendererHelper.saveProblemSource({
+            problemSource: req.body.problemSource,
+            writeFilePath: writeFilePath,
+        });
+
+        next(httpResponse.Ok('Loaded', {
+            filePath: result
+        }));
+    }));
+
+router.post('/question/editor/read',
+    authenticationMiddleware,
+    validate(readQuestionValidation),
+    // This is due to a typescript issue where the type mismatches extractMap
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    asyncHandler(async (req: RederlyExpressRequest<ReadQuestionRequest.params, unknown, ReadQuestionRequest.body, ReadQuestionRequest.query>, _res: Response, next: NextFunction) => {
+        if (_.isNil(req.session)) {
+            throw new Error(Constants.ErrorMessage.NIL_SESSION_MESSAGE);
+        }
+
+        const sourceFilePath = req.body.filePath;
+        const result = await rendererHelper.readProblemSource({
+            sourceFilePath: sourceFilePath
+        });
+
+        next(httpResponse.Ok('Loaded', {
+            problemSource: result
+        }));
+    }));
+
+router.post('/question/editor/catalog',
+    authenticationMiddleware,
+    validate(catalogValidation),
+    // This is due to a typescript issue where the type mismatches extractMap
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    asyncHandler(async (req: RederlyExpressRequest<CatalogRequest.params, unknown, CatalogRequest.body, CatalogRequest.query>, _res: Response, next: NextFunction) => {
+        if (_.isNil(req.session)) {
+            throw new Error(Constants.ErrorMessage.NIL_SESSION_MESSAGE);
+        }
+
+        const basePath = `private/my/${req.session.userId}`;
+        const result = await rendererHelper.catalog({
+            basePath: basePath,
+            // TODO what should the depth be?
+            // -1 did not do all
+            maxDepth: 100,
```

Comment:
I thought -1 was supposed to do the whole catalog

Reply:
I don't think I ever built that in. There is no "unlimited" search.

Reply:
Should maxDepth be passed in from the UI?

Reply:
In the future yes, the base path prefix is also hardcoded here
I feel like this makes it a little more secure in the meantime